src/core/public/application/application_service.tsx

```diff
@@ -304,14 +311,18 @@ export class ApplicationService {
         return absolute ? relativeToAbsolute(relUrl) : relUrl;
       },
       navigateToApp,
-      navigateToUrl: async (url) => {
+      navigateToUrl: async (url: string, { skipAppLeave }: NavigateToUrlOptions = {}) => {
```

Comment:
NIT: I'd go with explicit default here

```ts
navigateToUrl: async (url: string, { skipAppLeave = false }: NavigateToUrlOptions = {})
```

Reply:
I added [defaults](https://github.com/elastic/kibana/pull/128671/files#diff-94b600f79770d5081137b8938222dfc2718295a199b0685cd3fea4040114f451R316) for both flags.